src/lean_dojo/utils.py

```diff
@@ -143,6 +144,32 @@ def camel_case(s: str) -> str:
     """Convert the string ``s`` to camel case."""
     return _CAMEL_CASE_REGEX.sub(" ", s).title().replace(" ", "")
 
+def repo_type_of_url(url: str) -> str:
+    """Get the type of the repository.
+    
+    Args:
+        url (str): The URL of the repository.
+
+    Returns:
+        str: The type of the repository.
+    """
+    parsed_url = urlparse(url)
+    if parsed_url.scheme in ["http", "https"]:
+        # case 1 - GitHub URL
+        if "github.com" in url:
+            if not url.startswith("https://"):
+                logger.warning(f"{url} should start with https://")
+                return
+            else:
+                return "github"
+        # case 2 - remote Git URL
+        else:
+            return "remote"
+    # case 3 - local path
+    elif os.path.exists(parsed_url.path):
+        return "local"
+    else:
+        logger.warning(f"{url} is not a valid URL")
 
 @cache
 def get_repo_info(path: Path) -> Tuple[str, str]:
```

Comment:
`with working_directory(path)` is still needed in this function? There is no guarantee the CWD will be the same as `path` when calling it.

Reply:
I got the same error as @LuoKaiGSW and think this is the reason.